drivers/spi/spi_ll_stm32.c

```diff
@@ -129,6 +129,8 @@ static int spi_stm32_dma_tx_load(const struct device *dev, const uint8_t *buf,
 	/* give the client dev as arg, as the callback comes from the dma */
 	stream->dma_cfg.user_data = data;
 	/* pass our client origin to the dma: data->dma_tx.dma_channel */
+	stream->dma_cfg.linked_channel = STM32_DMA_HAL_OVERRIDE;
```

Comment:
`STM32_DMA_HAL_OVERRIDE` was designed specifically to be used with drivers using HAL with DMA APIs which conflicts with Zephyr DMA APIS (ex QSPI).
This is not the case here, so it should not be used.
If something specific to enable SPI/DMA for U5 is required it should be done under U5 specific flag and use other mecanism

Reply:
removed, not needed